exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControllerImpl.java

```diff
@@ -60,11 +62,13 @@ public ControlTunnel getTunnel(DrillbitEndpoint endpoint) {
     return new ControlTunnel(endpoint, connectionRegistry.getConnectionManager(endpoint));
   }
 
-  public void close() {
-    Closeables.closeQuietly(server);
-    for (ControlConnectionManager bt : connectionRegistry) {
-      bt.close();
+  public void close() throws IOException {
+    try {
+      Closeables.close(server, true);
+    } finally {
+      for (ControlConnectionManager bt : connectionRegistry) {
```

Comment:
If you are trying to make sure all resources are closed even if `server.close()` throws a runtime exception, then you are only solving half the problem. if `bt.close()` throws an exception then the loop may not proceed through the connection registry